draft-ietf-rats-corim.md

```diff
@@ -1066,6 +1066,182 @@ The following describes each member of the `concise-bom-tag` map.
+
+The selection process MUST yield at least one usable CoRIM.
+
+### CoBOMs Extraction
+
+All the available Concise Bill Of Material tags (CoBOMs) are collected from the selected CoRIMs.
+
+If any CoBOM is present then other tags in the CoRIM which are not activated by the CoBOM MUST be ignored.
+
+### Tags Identification and Validation
+
+Potentially valid tags -- i.e., Concise Module Identifiers (CoMID) and/or Concise Software Identifiers (CoSWID) and/or Concise Trust Anchor Stores (CoTS) -- are chosen from the selected CoRIMs.
+
+All the extracted tags MUST be successfully located and be syntactically and semantically valid.
+In particular, any internal cross-reference (e.g., CoMID-CoSWID linking triples) MUST be successfully resolved.
+
+### Appraisal Context Construction
+
+All of the potential tags are loaded into the Appraisal Context. The complete contents of each tag are loaded together, including Reference Values, Endorsed Values and cryptographic verification key material.
+
+This concludes the initialisation phase.
+
+## Evidence Collection
+
+In the evidence collection phase the verifier communicates with attesters to collect evidence. 
+
+The first part of the evidence collection phase does not perform any crypographic validation. This allows verifiers to use untrusted code for their initial evidence collection.
+
+The results of the evidence collection are protocol specific data and transcripts which can be processed by the verifier.
+
+### Cryptographic validation of Evidence
+
+If the authenticity of Evidence is secured by a cryptographic mechanism such as a signature, the first step in the Evidence Appraisal is to perform cryptographic validation of the Evidence.
+
+The exact cryptographic signature validation mechanics depend on the specific Evidence collection protocol.
+
+For example:
+In DICE, a proof of liveness is performed on the final key in the certificate chain. If this passes then a suitable certification path anchored on a trusted root certificate is looked up -- e.g., based on linking information obtained from the DeviceID certificate (see Section 9.2.1 of [DICE Layering Architecture](https://trustedcomputinggroup.org/wp-content/uploads/DICE-Layering-Architecture-r19_pub.pdf))-- in the Appraisal Context.  If found, then usual X.509 certificate validation is performed.
+In PSA, the verification public key is looked up in the appraisal context using the `euid` claim found in the PSA claims-set (see [Section 4.2.1 of I-D.tschofenig-rats-psa-token](https://www.ietf.org/archive/id/draft-tschofenig-rats-psa-token-10.html#name-instance-id).  If found, COSE Sign1 verification is performed accordingly.
+
+Independent of the specific method, the cryptographic integrity of Evidence MUST be successfully verified.
+
+> A CoRIM profile MUST describe:
+>
+> * How cryptographic verification key material is conveyed (e.g., using Attestation Keys triples, or CoTS tags)
```

Comment:
"conveyed" is misleading since RATS Arch uses 'conveyance' to mean how files such as manifests are communicated from one role / entity to another. This usage seems to be describing how verification key material is to be represented by Endorsements.